cli/src/semgrep/core_runner.py

```diff
@@ -668,20 +669,54 @@ def pprint(self, *, with_tables_for: RuleProduct) -> None:
             sep = "\n   "
             message = "No rules to run."
             if with_tables_for == RuleProduct.sca:
-                if auth.get_token() is None:
-                    message = sep.join(
-                        wrap(
-                            "💎 Sign in with `semgrep login` and run `semgrep ci` to find dependency vulnerabilities and advanced cross-file findings.",
-                            width=70,
+                """
+                We need to account for several edges cases:
+                 - `semgrep ci` was invoked but no rules were found (e.g. no lockfile).
+                 - `semgrep scan` was invoked with the supply-chain flag and no rules found.
+                 - `semgrep ci` was invoked without the supply-chain flag or feature enabled.
+                """
+                # 1. Validate that the user is indeed running SCA.
+                ctx = get_context()
```

Comment:
I have no idea why all this reporting stuff is in core_runner.py. This file is supposed to just contain code to interact with semgrep-core.
All of this should be in a separate file really (this is ok for this PR; just saying).

Reply:
If it was in another file we could also pass arguments to it, such as what was the command name, whether -supply-chain was passed instead of using globals and inspecting Click state.